lib/test/CDataFrameAnalysisSpecificationFactory.cc

```diff
@@ -272,97 +278,116 @@ std::string
         writer.Double(m_DownsampleFactor);
     }
     if (m_MaximumNumberTrees > 0) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::MAX_TREES);
+        writer.Key(TRunner::MAX_TREES);
         writer.Uint64(m_MaximumNumberTrees);
     }
     if (m_FeatureBagFraction > 0.0) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::FEATURE_BAG_FRACTION);
+        writer.Key(TRunner::FEATURE_BAG_FRACTION);
         writer.Double(m_FeatureBagFraction);
     }
     if (m_NumberRoundsPerHyperparameter > 0) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::MAX_OPTIMIZATION_ROUNDS_PER_HYPERPARAMETER);
+        writer.Key(TRunner::MAX_OPTIMIZATION_ROUNDS_PER_HYPERPARAMETER);
         writer.Uint64(m_NumberRoundsPerHyperparameter);
     }
     if (m_BayesianOptimisationRestarts > 0) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::BAYESIAN_OPTIMISATION_RESTARTS);
+        writer.Key(TRunner::BAYESIAN_OPTIMISATION_RESTARTS);
         writer.Uint64(m_BayesianOptimisationRestarts);
     }
     if (m_NumberTopShapValues > 0) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::NUM_TOP_FEATURE_IMPORTANCE_VALUES);
+        writer.Key(TRunner::NUM_TOP_FEATURE_IMPORTANCE_VALUES);
         writer.Uint64(m_NumberTopShapValues);
     }
     if (m_PredictionFieldName.empty() == false) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::PREDICTION_FIELD_NAME);
+        writer.Key(TRunner::PREDICTION_FIELD_NAME);
         writer.String(m_PredictionFieldName);
     }
-    if (m_PredictionFieldType.empty() == false) {
-        writer.Key(api::CDataFrameTrainBoostedTreeClassifierRunner::PREDICTION_FIELD_TYPE);
-        writer.String(m_PredictionFieldType);
-    }
-    if (analysis == classification()) {
-        writer.Key(api::CDataFrameTrainBoostedTreeClassifierRunner::NUM_CLASSES);
-        writer.Uint64(m_NumberClasses);
-        writer.Key(api::CDataFrameTrainBoostedTreeClassifierRunner::NUM_TOP_CLASSES);
-        writer.Uint64(m_NumberTopClasses);
-    }
     if (m_CustomProcessors.IsNull() == false) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::FEATURE_PROCESSORS);
+        writer.Key(TRunner::FEATURE_PROCESSORS);
         writer.write(m_CustomProcessors);
     }
     if (m_EarlyStoppingEnabled == false) {
-        writer.Key(api::CDataFrameTrainBoostedTreeRunner::EARLY_STOPPING_ENABLED);
+        writer.Key(TRunner::EARLY_STOPPING_ENABLED);
         writer.Bool(m_EarlyStoppingEnabled);
     }
 
+    if (analysis == classification()) {
+        writer.Key(TClassificationRunner::NUM_CLASSES);
+        writer.Uint64(m_NumberClasses);
+        writer.Key(TClassificationRunner::NUM_TOP_CLASSES);
+        writer.Uint64(m_NumberTopClasses);
+        if (m_PredictionFieldType.empty() == false) {
+            writer.Key(TClassificationRunner::PREDICTION_FIELD_TYPE);
+            writer.String(m_PredictionFieldType);
+        }
+        if (m_ClassificationWeights.size() > 0) {
```

Comment:
```suggestion
        if (m_ClassificationWeights..empty() == false) {
```
nit: for consistency with the code above and also to prevent clang-tidy readability complaints.

Reply:
Agreed. Done in [7809056](https://github.com/elastic/ml-cpp/pull/1725/commits/7809056cdf48bc2ddd5db8148da3e33d75cb4c39).